ext/fast_jsonparser/fast_jsonparser.cpp

```diff
@@ -27,9 +27,12 @@ static VALUE make_ruby_object(dom::element element)
             for (dom::key_value_pair field : dom::object(element))
             {
                 std::string_view view(field.key);
-                VALUE k = rb_intern(view.data());
-                VALUE v = make_ruby_object(field.value);
-                rb_hash_aset(hash, ID2SYM(k), v);
+                VALUE k = rb_utf8_str_new(view.data(), view.size());
+                if (symbolize_names) {
+                    k = ID2SYM(rb_intern_str(k));
```

Comment:
note that you merged the `rb_intern` change as well.

Reply:
@casperisfine thank you pointing it out. I missed it.

I am not aware of difference between `rb_intern` & `rb_intern_str` .

How it will affect fast_jsonparser?

Reply:
That's what makes the perf difference we talked about.

If you wish to be faster this part should be:

```c
                VALUE k;
                if (symbolize_names) {
                    k = ID2SYM(rb_intern2(view.data(), view.size()));
                } else {
                    k = rb_utf8_str_new(view.data(), view.size());
                }
```

Reply:
Thanks for clarification @casperisfine

Reply:
For now lets continue with utf8, we will have to update benchmark result and close PR #8